gunicorn/http/message.py

```diff
@@ -90,7 +90,10 @@ def parse_headers(self, data):
             if curr.find(":") < 0:
                 raise InvalidHeader(curr.strip())
             name, value = curr.split(":", 1)
-            name = name.rstrip(" \t").upper()
+            if self.cfg.strip_header_spaces:
+                name = name.rstrip(" \t").upper()
```

Comment:
shouldn't we return an error if it has "spaces" at the end? We need to check the spec about it.

Reply:
Well this is the option (disabled by default) to still it if some people really need it. The Normal case (in the else) will not strip the same, the regex after it will fail and yield an error.